### PR TITLE
Fix storage test sqlite bookmarks

### PR DESCRIPTION
**Pull request by fluffyemily**
Resolves:

    TestSQLLiteBookmarks: all tests

https://bugzilla.mozilla.org/show_bug.cgi?id=1336056

### CODE REVIEW

File: Storage/ThirdParty/SwiftData.swift [hunk 1]
```diff
@@ -267,6 +267,8 @@ private class SQLiteDBStatement {
             } else if obj is Date {
                 let timestamp = (obj as! Date).timeIntervalSince1970
                 status = sqlite3_bind_double(pointer, Int32(index+1), timestamp)
+            } else if obj is UInt64 {
```
Comment (sleroux) [inline Storage/ThirdParty/SwiftData.swift]: I ran into a similar issue when I was working on fixing another test case in Storage and it was actually being caused by an incorrect migration from NSNumber [1]. Could that also be the case here?

[1] https://github.com/mozilla-mobile/firefox-ios/commit/71bc3917d27f09a8806d279694368768933cd71f

Comment (sleroux) [inline Storage/ThirdParty/SwiftData.swift]: Note: Date.now() has always produced a `Timestamp` which is a UInt64 which makes me curious about if this change is needed.

Comment (fluffyemily) [inline Storage/ThirdParty/SwiftData.swift]: It strikes me as more Swifty to parse a `UInt64` directly than to use an `NSNumber` to add to an `Any` in order to convert a `UInt64` into a `Double`.

I went through this several times. If the obj is a UInt64, a Timestamp, then the value doesn't get bound. Then, when the statement is executed, an error is thrown becuase `last_modified` is a nil object. To be honest, I have no idea how this ever worked, if a previous version also passed in a UInt64.

Comment (sleroux) [inline Storage/ThirdParty/SwiftData.swift]: > It strikes me as more Swifty to parse a UInt64 directly than to use an NSNumber to add to an Any in order to convert a UInt64 into a Double.

Yup I agree

> To be honest, I have no idea how this ever worked, if a previous version also passed in a UInt64.

So I think what was happening before was that the NSNumber gets interpreted as a Double which essentially captures anything a number could be (float, long long, etc). 

I'll make sure to go back to my previous PR and update NSNumber conversion to UInt64 when this lands!

**Pull request merged**